test/cpp_extensions/open_registration_extension/pytorch_openreg/csrc/OpenRegMem.cpp

```diff
@@ -32,7 +32,7 @@ struct OpenRegAllocator final : at::Allocator {
   }
 
   static void ReportAndDelete(void* ptr) {
-    if (!ptr) {
+    if (!ptr || is_shutting_down()) {
```

Comment:
fwiw, we use this

```
  // Leak the pyobj if not initialized.  This can happen if we are running
  // exit handlers that are destructing tensors with residual (owned)
  // PyObjects stored in them.
  if (!Py_IsInitialized())
    return;
```

So you don't actually need the binding

Reply:
Awesome! I use this and it works!